test/functional/p2p_filter.py

```diff
@@ -135,14 +136,32 @@ def test_msg_mempool(self):
         self.log.info("Check that a node with bloom filters enabled services p2p mempool messages")
         filter_peer = P2PBloomFilter()
 
-        self.log.debug("Create a tx relevant to the peer before connecting")
-        txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=9 * COIN)["txid"]
+        self.log.info("Create two tx before connecting, one relevant to the node another that is not")
+        rel_txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=1 * COIN)["txid"]
+        irr_txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=getnewdestination()[1], amount=2 * COIN)["txid"]
+        irr_wtxid = self.nodes[0].getmempoolentry(irr_txid)["wtxid"]
 
-        self.log.debug("Send a mempool msg after connecting and check that the tx is received")
+        self.log.info("Send a mempool msg after connecting and check that the relevant tx is received")
         self.nodes[0].add_p2p_connection(filter_peer)
-        filter_peer.send_and_ping(filter_peer.watch_filter_init)
-        filter_peer.send_message(msg_mempool())
-        filter_peer.wait_for_tx(txid)
+        filter_peer.send_message(filter_peer.watch_filter_init)
+        filter_peer.send_and_ping(msg_mempool())
+        filter_peer.wait_for_tx(rel_txid)
+
+        self.log.info("Request the irrelevant transaction even though it has not being offered to us")
+        filter_peer.tx_received = False
+        filter_peer.send_and_ping(msg_getdata([CInv(t=MSG_WTX, h=int(irr_wtxid, 16))]))
+        self.log.info("We should get it anyway given it was on the mempool when the MEMPOOL message was sent")
+        assert filter_peer.tx_received
+
+        self.log.info("Create a third transaction (that was therefore not part of the mempool message) and try to retrieve it")
+        filter_peer.tx_received = False
+        add_txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=getnewdestination()[1], amount=3 * COIN)["txid"]
+        # sync_with_ping forces the node to trickle, which increases m_last_inv_sequence
+        filter_peer.sync_with_ping()
+        filter_peer.send_and_ping(msg_getdata([CInv(t=MSG_WTX, h=int(self.nodes[0].getmempoolentry(add_txid)["wtxid"], 16))]))
+        self.log.info("We should be able to request this one given too since we force the node to trickle")
+        assert filter_peer.tx_received
```

Comment:
What's the purpose of this test? It seems like we're just testing that we don't delay announcements to peers with noban permissions. That doesn't have anything to do with bloom filters or the mempool message.

Assuming the original intent was to check that they won't be able to request the tx before announcement, you could remove noban permissions and assert the opposite, but I think that's covered in other tests already.

Reply:
I guess we could get rid of this bit. 

The purpose was really to test how the transaction won't be available until the node has trickled for the requesting peer, but that is not something we can check without introducing a time-variant to the test that will make it fail sparsely.